src/pip/_internal/utils/misc.py

```diff
@@ -510,6 +523,12 @@ def unzip_file(filename, location, flatten=True):
                 fn = split_leading_dir(name)[1]
             fn = os.path.join(location, fn)
             dir = os.path.dirname(fn)
+            # https://github.com/pypa/pip/issues/3907
```

Comment:
Normally we don't link to issue numbers like this in the code base. I would just say something like, "Prevent writing outside the target directory."